tests/compiletest.rs

```diff
@@ -13,16 +13,15 @@ macro_rules! eprintln {
     }
 }
 
-#[cfg(debug_assertions)]
-const MIRI_PATH: &str = "target/debug/miri";
-#[cfg(not(debug_assertions))]
-const MIRI_PATH: &str = "target/release/miri";
+fn miri_path() -> String {
+    format!("target/{}/miri", env!("PROFILE"))
```

Comment:
You can do this with `const` and `concat!("target/", env!("PROFILE"), "/miri")`.